src/packet_analysis/Manager.cc

```diff
@@ -18,64 +13,66 @@ Manager::Manager()
 	{
 	}
 
-Manager::~Manager()
-	{
-	}
-
 void Manager::InitPostScript()
 	{
-	auto analyzer_mapping = zeek::id::find("PacketAnalyzer::config_map");
+	// Instantiate objects for all available analyzers
+	for ( const auto& analyzerComponent : GetComponents() )
+		{
+		if ( AnalyzerPtr newAnalyzer = InstantiateAnalyzer(analyzerComponent->Tag()) )
+			analyzers.emplace(analyzerComponent->Name(), newAnalyzer);
+		}
+
+	// Read in analyzer map and create dispatchers
+	auto& analyzer_mapping = zeek::id::find("PacketAnalyzer::config_map");
 	if ( ! analyzer_mapping )
 		return;
 
 	auto mapping_val = analyzer_mapping->GetVal()->AsVectorVal();
 	if ( mapping_val->Size() == 0 )
 		return;
 
-	Config configuration;
 	for (unsigned int i = 0; i < mapping_val->Size(); i++)
 		{
 		auto* rv = mapping_val->At(i)->AsRecordVal();
-		auto parent = rv->GetField("parent");
-		std::string parent_name = parent ? Lookup(parent->AsEnumVal())->Name() : "ROOT";
+		//TODO: Make that field a string for usability reasons
+		//TODO: Check error handling when fields are omitted
+		auto& parent_tag = rv->GetField("parent");
+		std::string parent_name = parent_tag ? Lookup(parent_tag->AsEnumVal())->Name() : "ROOT";
 		auto identifier = rv->GetField("identifier")->AsCount();
-		auto analyzer = rv->GetField("analyzer")->AsEnumVal();
+		auto analyzer_tag = rv->GetField("analyzer")->AsEnumVal();
+		auto analyzer_name = Lookup(analyzer_tag)->Name();
 
-		configuration.AddMapping(parent_name, identifier, Lookup(analyzer)->Name());
-		}
-
-	// Instantiate objects for all analyzers
-	for ( const auto& current_dispatcher_config : configuration.GetDispatchers() )
-		{
-		for ( const auto& current_mapping : current_dispatcher_config.GetMappings() )
+		if ( analyzers.find(analyzer_name) == analyzers.end() )
 			{
-			// Check if already instantiated
-			if ( analyzers.count(current_mapping.second) != 0 )
-				continue;
+			reporter->InternalWarning("Mapped analyzer %s not found.", analyzer_name.c_str());
+			continue;
+			}
 
-			// Check if analyzer exists
-			if ( AnalyzerPtr newAnalyzer = InstantiateAnalyzer(current_mapping.second) )
-				analyzers.emplace(current_mapping.second, newAnalyzer);
+		if ( parent_name == "ROOT" )
+			{
+			root_dispatcher.Register(identifier, analyzers[analyzer_name]);
+			continue;
 			}
-		}
 
-	// Generate Dispatchers, starting at root
-	root_dispatcher = GetDispatcher(configuration, "ROOT");
-	if ( root_dispatcher == nullptr )
-		reporter->InternalError("No dispatching configuration for ROOT of packet_analysis set.");
+		if ( analyzers.find(parent_name) == analyzers.end() )
+			{
+			reporter->InternalWarning("Parent analyzer %s not found.", parent_name.c_str());
+			continue;
+			}
 
-	// Set up default analysis
-	auto it = analyzers.find("DefaultAnalyzer");
-	if ( it != analyzers.end() )
-		default_analyzer = it->second;
-	else
-		default_analyzer = InstantiateAnalyzer("DefaultAnalyzer");
+		auto& parent_analyzer = analyzers[parent_name];
```

Comment:
This causes an extra lookup into the map. I know it's only done at startup, but you can just use the result from the `find()` above instead of looking it up a second time.

Reply:
You're completely right. I changed that in both places...